### DISCUSSION TITLE
How to test `url` column? (test that content is displayed)

### DISCUSSION THREAD

**Discussion by eithed** [filamentphp/filament#8188] [Help]
Hello!

I'm using `url` column as such:
```
Tables\Columns\TextColumn::make('url')
    ->url(fn (Record $record): ?string => $record->foo ? $record->bar : null)
```

And I'd like to write tests for this. How can I do so?

---

I was able to create a workaround, ie:
```
Tables\Columns\TextColumn::make('url')
    ->getStateUsing(fn (Record $record): ?string => $record->foo ? sprintf("<a href='%s' class='text-primary-600'>%s</a>", $record->bar, $record->url) : null)
    ->html(),
```
and then test using `assertTableColumnStateSet`, but this suffers from:
- if implementation of url render changes, I need to update what I'm outputting
- I need to be testing that entire `sprintf("<a href='%s' class='text-primary-600'>%s</a>", $record->bar, $record->url)` is set as a state, while really I'd like to be testing existence (or inexistence) of a record with label url, pointing to bar

**Comment by pboivin** [comment]
Hi @eithed, just a thought: do you think `assertSee()` or `assertSeeHtml()` could cover what you're trying to test?

**Reply by eithed** [reply]
I think `assertHtml` suffers from the same issue as setting a state to certain html and then testing for that html - if implementation of `url()` changes (for example - an additional css class is added natively in Filament), the tests need to change (when, I'd imagine, if `assertTableColumnUrl` were to exist, it would require parity with `url()`).

Thinking about it if I really want to test that there's a record in output table that has certain url (test for `sprintf("<a href='%s'", $record->url())`) and certain text label (test for `bar`), so indeed two `assertSee` to explicitly test that could suffice
